internal/plugin/grpc_client_plugin.go

```diff
@@ -230,3 +325,18 @@ func (gc *GrpcClient) sendDataPlaneStatusUpdate(
 
 	return err
 }
+
+type GrpcConfigClient struct {
+	grpcOverviewFn    func(ctx context.Context, request *v1.GetOverviewRequest) (*v1.FileOverview, error)
+	grpFileContentsFn func(ctx context.Context, request *v1.GetFileRequest) (*v1.FileContents, error)
+}
```

Comment:
Types have generally been defined at the top of the file in our code (even though I totally get why you have this here). I'm open to having a discussion on changing this convention.

Reply:
Ya Donal had this part of the code started and had it here. I was originally going to move it but then also got why it was here so left it,  discussion might be good to get others thoughts as I understand both options

Reply:
I prefer it at the top of the file personally. It's weird seeing struct definitions sandwiched between functions

Reply:
It's also weird that you have random method definitions nowhere near the struct they are attached to. They are logically related to one another, so to me it makes sense to have them grouped up.

Perhaps we should see how other Go projects handle this.

Reply:
Couldn't find any guidelines for this. The Uber guide (see [here](https://github.com/uber-go/guide/blob/master/style.md#function-grouping-and-ordering)) does call out the order as `struct` definition -> constructor -> methods -> other functions but it does not fully answer our question here.

I was reading through the `opentelemetry-collector-contrib` repository for examples and in all the examples I found, these were split into their own files. Should we possibly move `GrpcConfigClient` into its own file?